### PR TITLE
add long lived cluster management job and e2e test

### PR DESCRIPTION
**Pull request by bparees**
related to: https://jira.coreos.com/browse/DPP-1338
blocked by: https://jira.coreos.com/browse/DPP-2164

### CODE REVIEW

**Comment by bparees** [comment]
status update:  i think the only thing left to be done on this is to switch to using llc aws creds.

once that's done this PR will introduce a job that will on a daily basis:

1) check if the existing long lived cluster is healthy
2) if not, perform must-gather and attempt to tear it down
3) if the cluster wasn't healthy, install a new cluster

the job will report failed if the cluster was not healthy.

next immediate steps (additional PRs) will be to run e2e against the cluster periodically, and introducing a long-lived workload with healthchecking.

**Comment by bparees** [comment]
@smarterclayton @stevekuznetsov i think this is ready to go, for 4.1 at least..  once i see it working for 4.1 i'll introduce 4.2+4.3 versions.  Can you review?

File: ci-operator/jobs/openshift/release/openshift-release-release-4.1-periodics.yaml [hunk 1]
```diff
@@ -1,4 +1,232 @@
...
+  decorate: true
+  labels:
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: longlived-aws-cluster-maintenance-4.1
+  spec:
+    containers:
+    - command:
+      - /bin/bash
+      - -c
+      - "#!/bin/bash\nset -euo pipefail\n\nexport PATH=$PATH:/tmp/bin\nmkdir /tmp/bin\ncurl
```
Comment (smarterclayton) [inline ci-operator/jobs/openshift/release/openshift-release-release-4.1-periodics.yaml]: Fix this formatting

File: ci-operator/templates/openshift/longlived/longlived-test.yaml [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        set -euo pipefail
+
+        export PATH=/usr/libexec/origin:$PATH
+
+        trap 'touch /tmp/done' EXIT
+        trap 'kill $(jobs -p); exit 0' TERM
+
+        mkdir -p "${HOME}"
+
+        until oc --insecure-skip-tls-verify wait clusterversion/version --for condition=available 2>/dev/null; do
```
Comment (smarterclayton) [inline ci-operator/templates/openshift/longlived/longlived-test.yaml]: This is the installers job now, you shouldn't have to do this.

File: ci-operator/templates/openshift/longlived/longlived-test.yaml [hunk 3]
```diff
@@ -0,0 +1,156 @@
+kind: Template
```
Comment (smarterclayton) [inline ci-operator/templates/openshift/longlived/longlived-test.yaml]: The directory and file naming feels weird.  But is fine for now

File: ci-operator/jobs/openshift/release/openshift-release-release-4.1-periodics.yaml [hunk 4]
```diff
@@ -1,4 +1,232 @@
...
+        secretName: ci-pull-credentials
+    - name: longlived-aws-cluster-credentials
+      secret:
+        secretName: longlived-aws-cluster-credentials
+- agent: kubernetes
+  cron: 0 12 * * 1-5
+  decorate: true
+  labels:
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: longlived-aws-cluster-e2e-4.1
```
Comment (smarterclayton) [inline ci-operator/jobs/openshift/release/openshift-release-release-4.1-periodics.yaml]: The name pattern seems weird.

`longlived-e2e-aws-4.1` is probably more correct.  And `longlived-maintenance-aws-4.1`.

I'd also prefer `endurance-` or `stability-` for the naming for all of this.  `longlived` lacks something.  `longrunning` is hard to understand.  We will have `soak` and `performance` variants eventually.

File: ci-operator/jobs/openshift/release/openshift-release-release-4.1-periodics.yaml (hunk 1 above)
Comment (bparees) [reply ci-operator/jobs/openshift/release/openshift-release-release-4.1-periodics.yaml]: I can't. Make jobs does this.

Comment (bparees) [reply ci-operator/jobs/openshift/release/openshift-release-release-4.1-periodics.yaml]: context: https://github.com/openshift/release/commit/ef3cd1d0c5a65fc78dc9b4a4d07d536d983554f1#r35370176

(you've argued with @stevekuznetsov about this before, it appears to have gone unresolved)

File: ci-operator/templates/openshift/longlived/longlived-test.yaml (hunk 2 above)
Comment (bparees) [reply ci-operator/templates/openshift/longlived/longlived-test.yaml]: will remove

File: ci-operator/jobs/openshift/release/openshift-release-release-4.1-periodics.yaml (hunk 4 above)
Comment (bparees) [reply ci-operator/jobs/openshift/release/openshift-release-release-4.1-periodics.yaml]: going with endurance-<jobtype>-<clustertype>

**Comment by bparees** [comment]
@smarterclayton comments addressed except for the formatting thing that i don't think i can fix.

File: ci-operator/templates/openshift/longlived/longlived-test.yaml (hunk 3 above)
Comment (bparees) [reply ci-operator/templates/openshift/longlived/longlived-test.yaml]: i renamed it to endurance. i'm sure it's still weird but at it's consistent anyway.

File: ci-operator/jobs/openshift/release/openshift-release-release-4.1-periodics.yaml (hunk 1 above)
Comment (smarterclayton) [reply ci-operator/jobs/openshift/release/openshift-release-release-4.1-periodics.yaml]: Make jobs is a turd if it breaks human entered text.  Can you open a card for DPTP?  Needs to get fixed or disabled for these jobs.

Comment (bparees) [reply ci-operator/jobs/openshift/release/openshift-release-release-4.1-periodics.yaml]: submitted a ticket to openshift-ci-requsts

Comment (bparees) [reply ci-operator/jobs/openshift/release/openshift-release-release-4.1-periodics.yaml]: https://jira.coreos.com/browse/DPTP-658

**Pull request merged**